src/c++/perf_analyzer/test_command_line_parser.cc

```diff
@@ -524,6 +524,36 @@ TEST_CASE("Testing Command Line Parser")
     }
   }
 
+  SUBCASE("Option : --sequence-length-variation")
+  {
+    SUBCASE("non-negative")
+    {
+      int argc = 5;
+      char* argv[argc] = {app_name, "-m", model_name, "--sequence-length",
+                          "0.3"};
+
+      REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+      CHECK(!parser.UsageCalled());
+
+      exp->sequence_length_variation = 0.3;
+    }
+
+    SUBCASE("negative")
+    {
+      int argc = 5;
+      char* argv[argc] = {app_name, "-m", model_name,
+                          "--sequence-length-variation", "-0.1"};
+
+      REQUIRE_NOTHROW(act = parser.Parse(argc, argv));
+      REQUIRE(parser.UsageCalled());
+      CHECK_STRING(
+          "Usage Message", parser.GetUsageMessage(),
+          "sequence length variation must be positive");
+
+      exp->sequence_length_variation = -0.1;
+    }
+  }
+
```

Comment:
Do you think it would be possible to add a test case to confirm the round robin order? For example, if sequence length is 4 and number of entries is 3, the data stream indexes would be ({0, 1, 2}, {1, 2, 0}, ...).

Reply:
This is what the [update_seq_json_data: testing the UpdateSeqJsonData function](https://github.com/triton-inference-server/client/blob/matthewkotila-sequence-length/src/c%2B%2B/perf_analyzer/test_infer_context.cc#L50) test does.

The round-robin-ness is guaranteed by the modulus operator [here](https://github.com/triton-inference-server/client/blob/matthewkotila-sequence-length/src/c%2B%2B/perf_analyzer/infer_context.cc#L195). The above linked test sets up a situation where a sequence has length 5, and there are 4 inputs in the JSON, and there is 1 remaining query, meaning 4 requests have already been sent, and 4 inputs from the JSON have already been used, in order. Which means the next request, the fifth and final request of the sequence, would have to loop back around from the last input in the JSON back to the first.

The test confirms this by showing that the resulting used step id is `0`.